File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if(root==null) return new ArrayList<>();
        List<Integer> ans = new ArrayList<>();
        
        //Create que for level order traversal
        LinkedList<TreeNode> que = new LinkedList<>();
        que.addLast(root);
        //Begin BFS
        while(que.size()!=0) {
            int size = que.size();
            //Inner while loop for each level
            while(size-->0) {
                TreeNode removeN = que.removeFirst();
                //SETUP next level in que
                if(removeN.left !=null) {
                    que.addLast(removeN.left);
                }
                
                if(removeN.right !=null) {
                    que.addLast(removeN.right);
                }
                //Add last node of this level in our ans
                if(size==0) ans.add(removeN.val);
            }
        }
        return ans;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if(root == nullptr) return {};
        vector<int> ans;
        //setup que for BFS
        queue<TreeNode*> que;
        que.push(root);
        
        //start bfs
        while(que.size()!=0) {
            int size = que.size();
            //Inner loop for every level
            while(size-->0) {
                TreeNode* removeN = que.front(); que.pop();
                
                if(removeN->left) que.push(removeN->left);
                
                if(removeN->right) que.push(removeN->right);
                
                
                //last node save in ans;
                if(size==0) ans.push_back(removeN->val);
            }
        }
        return ans;
    }
};